docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,550 @@
+    "\n",
+    "- **With Repo:** Whether a function requires additional files or dependencies within a Git repo or archive file. This Git repo or archive file is specified on a project level via `project.set_source(...)`, which the function consumes. If this parameter is ommited, the default is `False`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "edeffa8d",
+   "metadata": {},
+   "source": [
+    "<a id=\"building-images\"></a>\n",
+    "### Building Images"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "954c7f23",
+   "metadata": {},
+   "source": [
+    "If your MLRun function requires additional libraries or files, you may need to build a new Docker image. You can do this by specifying a base image using as the `image`, your requirements via `requirements`, and (optionally) your source code via `with_repo=True` (where the source is specified by `project.set_source(...)`). See [**Build function image**](./image-build.html) for more information on when a build is required.\n",
+    "``` {admonition} Note\n",
+    "When using `with_repo`, the contents of the Git repo or archive are available in the current working directory of your MLRun function during runtime.\n",
+    "```\n",
+    "\n",
+    "A good place to start is one of the default MLRun images:\n",
+    "- `mlrun/mlrun`: Suits most lightweight components (includes `sklearn`, `pandas`, `numpy` and more)\n",
+    "- `mlrun/ml-models`: Suits most CPU ML/DL workloads (includes `Tensorflow`, `Keras`, `PyTorch` and more)\n",
+    "- `mlrun/ml-models-gpu`: Suits most GPU ML/DL workloads (includes GPU `Tensorflow`, `Keras`, `PyTorch` and more )\n",
+    "\n",
+    "Dockerfiles for the MLRun images can be found [**here**](https://github.com/mlrun/mlrun/tree/development/dockerfiles)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8e827478",
+   "metadata": {},
+   "source": [
+    "<a id=\"single-source-file\"></a>\n",
+    "### Single Source File"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "be147543",
+   "metadata": {},
+   "source": [
+    "The simplest way to create a function is to use a single file as the source. The code itself is embedded into the MLRun function object. This makes the function quite portable as it does not depend on any external files. You can use any source file supported by MLRun such as Python or Jupyter notebook.\n",
+    "``` {admonition} Note\n",
+    "MLRun is not limited to Python. Bash, Go, etc. files are also supported.\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "41f62728",
+   "metadata": {},
+   "source": [
+    "#### Python"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bf09c489",
+   "metadata": {},
+   "source": [
+    "This is the simplest way to create a function out of a given piece of code - simply pass in the path to the Python file *relative to your project context directory*"
```

Comment:
"This is the simplest way to create a function out of a given piece of code. Simply pass in the path to the Python file *relative to your project context directory*."